hal/api/RawSerial.h

```diff
@@ -32,6 +32,8 @@ namespace mbed {
  * Can be used for Full Duplex communication, or Simplex by specifying
  * one pin as NC (Not Connected)
  *
+ * @Note Synchronization level: Not protected
```

Comment:
Good catch. I removed the mutex.  The "Set by subclass" was only marked on abstract classes, so that is why I used "Not Protected" on RawSerial.